game-core/src/main/java/games/strategy/triplea/ai/pro/simulate/ProSimulateTurnUtils.java

```diff
@@ -181,28 +189,34 @@ public static void simulateBattles(
       for (final Unit u : moveMap.get(fromTerritory).getUnits()) {
         if (!amphibUnits.contains(u)) {
           final Unit toUnit = transferUnit(u, unitTerritoryMap, usedUnits, toData, player);
-          patd.addUnit(toUnit);
-          ProLogger.trace("---Transferring unit " + u + " to " + toUnit);
+          if (toUnit != null) {
```

Comment:
not necessary, but any thoughts to using `Optional` to avoid the intermediate variable?

 transferUnit(u, unitTerritoryMap, usedUnits, toData, player).ifPresent(
     unit -> {
             :
             :
      });

Optional return values generally work out very nicely and avoid needing to notice there is a nullable annotation and forces you (type safety) to deal with the null.